hopsworks-common/src/main/java/io/hops/hopsworks/common/kafka/KafkaController.java

```diff
@@ -75,7 +94,297 @@
   private CertsFacade userCerts;
   @EJB
   private Settings settings;
-
+  @EJB
+  private ProjectFacade projectFacade;
+  @EJB
+  private UserFacade userFacade;
+  
+  public void createTopic(Project project, TopicDTO topicDto) throws KafkaException,
+    ProjectException, UserException, InterruptedException, ExecutionException {
+    
+    if (topicDto == null) {
+      throw new IllegalArgumentException("topicDto was not provided.");
+    }
+    
+    String topicName = topicDto.getName();
+    
+    if (kafkaFacade.findTopicByName(topicDto.getName()).isPresent()) {
+      throw new KafkaException(RESTCodes.KafkaErrorCode.TOPIC_ALREADY_EXISTS, Level.FINE, "topic name: " + topicName);
+    }
+    
+    if (kafkaFacade.findTopicsByProject(project).size() > project.getKafkaMaxNumTopics()) {
+      throw new KafkaException(RESTCodes.KafkaErrorCode.TOPIC_LIMIT_REACHED, Level.FINE,
+        "topic name: " + topicName + ", project: " + project.getName());
+    }
+    
+    SchemaTopics schema =
+      kafkaFacade.findSchemaByNameAndVersion(topicDto.getSchemaName(), topicDto.getSchemaVersion()).orElseThrow(() ->
+        new KafkaException(RESTCodes.KafkaErrorCode.SCHEMA_NOT_FOUND, Level.FINE, "topic: " + topicName));
+    
+    //check if the replication factor is not greater than the
+    //number of running brokers
+    try {
+      Set<String> brokerEndpoints = settings.getBrokerEndpoints();
```

Comment:
@kai-chi can we use `getTopicDetailsFromKafkaCluster` instead?

Reply:
no, `getTopicDetailsFromKafkaCluster` returns details of a topic, while here you are creating a new topic